lib/util/callbacks.ts

```diff
@@ -0,0 +1,51 @@
+import * as Bluebird from 'bluebird';
+import { Dictionary } from '../../typings/utils';
+
+// Use with: `findCallback(arguments)`.
+const findCallback = (args: IArguments): ((...args: any[]) => any) | void => {
+	const lastArg = args[args.length - 1];
+	if (typeof lastArg === 'function') {
+		return lastArg;
+	}
+
+	return;
+};
+
+const isThenable = (a: any): a is PromiseLike<any> => {
+	return typeof a === 'object' && typeof a.then === 'function';
+};
+
+const addCallbackSupport = <T extends (...args: any[]) => any>(fn: T): T => {
+	return function() {
+		const callback = findCallback(arguments);
+		const result = fn.apply(this, arguments);
+		if (!callback || !isThenable(result)) {
+			return result;
+		}
+
+		const bluebirdPromise =
+			'asCallback' in result
+				? (result as Bluebird<any>)
+				: Bluebird.resolve(result);
+
+		return bluebirdPromise.asCallback(callback);
+	} as T;
+};
+
+export const addCallbackSupportToModule = <T extends Dictionary<any>>(
+	sdkModule: T,
+): T => {
+	const result = {} as T;
```

Comment:
```suggestion
	const result: T = {};
```

Reply:
TS would complain since the assigned `{}` is missing atm all the props of T and that's why I needed the looser `as T`.
Otherwise we could do something:
```ts
const result: Partial<T> = {};
// for...
return result as T;
```
but I decided to stick with the approach of this PR since it had less type related code.

Reply:
ok